test/tests.js

```diff
@@ -230,6 +230,88 @@ test('dynamic queue', function(assert) {
   });
 });
 
+test('state machine', function(assert) {
+  var done = assert.async();
+  var stepGen = function(obj) {
+    swal.setDefaults({
+      input: 'select',
+      inputValidator: function(value) {
+        switch (value) {
+          case 'insert':
+            obj.insert('Step 2');
+            break;
+          case 'fork':
+            obj.fork(['Step 3', 'Step 4']);
+            break;
+          case 'repeatCurrent':
+            obj.repeatCurrent();
+            break;
+          case 'terminate':
+            obj.terminate();
+            break;
+          default:
+            break;
+        }
+        return new Promise(function(resolve) {
+          resolve();
+        });
+      }
+    });
+    switch (obj.current) {
+      case 'Step 1':
+        return {
+          title: 'Step 1',
+          inputOptions: { insert: 1 }
+        };
+      case 'Step 2':
+        return {
+          title: 'Step 2',
+          inputOptions: { fork: 1 }
+        };
+      case 'Step 4':
+        return {
+          title: 'Step 4',
+          inputOptions: { repeatCurrent: 1 }
+        };
+      case 'Step 5':
+        return {
+          title: 'Step 5',
+          inputOptions: { terminate: 1 }
+        };
+      default:
+        return obj.current;
+    }
+  };
+  swal.queue(stepGen, ['Step 1', 'Step 5']);
+  assert.equal('Step 1', $('.swal2-modal h2').text());
+  swal.clickConfirm();
+  setTimeout(function() {
+    assert.equal('Step 2', $('.swal2-modal h2').text());
```

Comment:
during fixing tests please take a look at https://github.com/limonte/sweetalert2/commit/736723e179487bc4c63e112d0c5ab561e7c710c0 ad change order of params in `equal()` functions - actual should be first param and expected should be second.